@joeyAghion => Consolidate all metaphysics interactions into one shared component for desktop + mobile (and use POST)

mzikherman opened the pull request:
cc @alloy 

This has the benefit of greatly reducing the Metaphysics papertrail log which is full of giant URLs.

The superagent lib and resolving the promise with the response data is virtually identical for GETs or POSTs. Literally the only difference is using `.set(...)` for POST vs `.query(...)` for GET, which is already handled.

Switched the default and tested some pages, all looks fine. Headers are being passed-over, etc.

We can wait a bit to examine the logs, and for fallout, before going in and cleaning out the ability to even use GET, if we want. But no rush on that.

Now actually in the body:
<img width="101" alt="screen shot 2017-03-08 at 5 57 16 pm" src="https://cloud.githubusercontent.com/assets/1457859/23728239/c9f3700c-0429-11e7-908a-4d3b8d0e3c04.png">

joeyAghion commented:
Muchas gracias! A 4 character change that fixes logs, makes inspecting network requests easier, _and_ is less prone to errors?

My only question is why make it an argument at all. Do any callers override the `method`?

mzikherman commented:
Good point @alloy !

Taking a closer look, I'll just go in and consolidate all the metaphysics code into one place. Shouldn't be too hard of a find + replace.

Then since I'll be modifying all those locations I can go in and remove the whole `method` thing. I was saying there's no huge rush to remove support for that, since you currently _can_ do that, but the only overrides are callers wanting to POST.

mzikherman commented:
cc @craigspaeth since kind of a big diff!

Ok this should be ready to go.

The commits tell the story, but basically:

- moved Metaphysics wrapper to top-level `lib/`, and it now only supports POST
- updated all `desktop/` and `mobile/` calls to point to new location (ugh relative paths!)
- deleted very similar metaphysics wrappers from both desktop and mobile
- minor test updates

I also browsed a bunch of major pages on the site in both desktop and mobile, doesnt look broken 👍

mzikherman commented on desktop/apps/artist/client/views/articles.coffee:
```diff
@@ -5,7 +5,7 @@ ArtworkRailView = require '../../../../components/artwork_rail/client/view.coffe
 sd = require('sharify').data
 query = require '../../queries/articles.coffee'
-metaphysics = require '../../../../lib/metaphysics.coffee'
+metaphysics = require '../../../../../lib/metaphysics.coffee'
```
All of these changes to the relative paths had to go up one level everywhere, since it moved from a root-level `lib/` folder within desktop and mobile, to project-level `lib/` (up one from the desktop/mobile location).

joeyAghion commented:
Thanks so much for making the larger refactor. If everyone jumps in like this, the desktop/mobile distinction will disappear quickly.
